### PR TITLE
scrubber: add remote_storage based listing APIs and use them in find-large-objects

### PR DESCRIPTION
**Pull request by arpad-m**
Add two new functions `stream_objects_with_retries` and `stream_tenants_generic` and use them in the `find-large-objects` subcommand, migrating it to `remote_storage`.

Also adds the `size` field to the `ListingObject` struct.

Part of #7547

### CODE REVIEW

**Review by skyzh — APPROVED** [review]
LGTM. It's worth noting that `list_streaming` could be an infinite stream if there is an error. The user of this stream should control the number of `next` calls and stop if there is a repeated error.

**Comment by arpad-m** [comment]
I have pushed two commits for yielding errors and ending the stream. Note that we already would end the stream when encountering an error: retryable errors would be retried but any other errors we'd yield immediately, and higher up the chain we'd exit at any error immediately (retried errors would not show up if a later attempt is successful).

File: storage_scrubber/src/lib.rs [hunk 1]
```diff
@@ -124,7 +126,7 @@ impl S3Target {
     pub fn with_sub_segment(&self, new_segment: &str) -> Self {
         let mut new_self = self.clone();
         if new_self.prefix_in_bucket.is_empty() {
-            new_self.prefix_in_bucket = format!("/{}/", new_segment);
+            new_self.prefix_in_bucket = format!("{}/", new_segment);
```
[the remark this answers is not in the document]
Comment (arpad-m) [reply storage_scrubber/src/lib.rs:129]: Hmmm the issue is that `RemotePath::new` doesn't want absolute paths.

[the remark this answers is not in the document]
Comment (arpad-m) [reply storage_scrubber/src/lib.rs]: This problem caused me to abandon `S3Target` usage in the first place, but I think I found a solution, just remove the leading `/` before converting to `RemotePath`.

**Pull request merged**